Lombiq.Analyzers.PowerShell/Rules/Measure-VariableNameFormat/Measure-VariableNameFormat.psm1

```diff
@@ -0,0 +1,344 @@
+                            "'$correctedParameterName' to start with an uppercase letter."
+                        ) -join ' '
+                    )
+
+                    $suggestedCorrections = New-Object System.Collections.ObjectModel.Collection[$correctionTypeName]
+                    $suggestedCorrections.add($correctionExtent) | Out-Null
+
+                    $analyzerViolations += [Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord]@{
+                        'Extent' = $parameter.Name.Extent
+                        'Message' = @(
+                            "Parameter names should start with an uppercase letter: '$parameterName' should be"
+                            "'$correctedParameterName'."
+                        ) -join ' '
+                        'RuleName' = 'PSUseCorrectParameterNameCasing'
+                        'RuleSuppressionID' = 'PSUseCorrectParameterNameCasing'
+                        'Severity' = 'Warning'
+                        'SuggestedCorrections' = $suggestedCorrections
+                    }
+                }
+            }
+
+            # Set up a new dictionary that contains the parameters of the functions with their parents (and root)
+            # included.
+            $functionParameterNamesWithParents = @{}
+            $functionParameterNamesWithParents[$rootIndex] = $functionParameterNames[$rootIndex]
+            foreach ($function in $functions)
+            {
+                $parentFunctions = Find-AstParents -AstObject $function -ParentType ([FunctionDefinitionAst])
+
+                # Add the parameters of the function itself.
+                $functionParameterNamesWithParents[$function.Name] += $functionParameterNames[$function.Name]
+                # Add the parameters of the parent functions.
+                foreach ($parentFunction in $parentFunctions)
+                {
+                    $functionParameterNamesWithParents[$function.Name] += $functionParameterNames[$parentFunction.Name]
+                }
+                # Add the parameters of the root.
+                $functionParameterNamesWithParents[$function.Name] += $functionParameterNames[$rootIndex]
+            }
+
+            # Extract each variable expression from the whole AST.
+            $variables = $Ast.FindAll(
+                {
+                    param([Ast] $astObject)
+                    return ($astObject -is [VariableExpressionAst])
+                },
+                $true
+            )
+
+            foreach ($variable in $variables)
+            {
+                $variableName = $variable.Extent.Text
+
+                # Due to the added complexity vs. their usage frequency, skip scoped variables and path-like
+                # expressions. Also skipping environment variables, where the casing doesn't matter.
+                if ($variableName.Contains(':'))
+                {
+                    continue
+                }
```

Comment:
Since this cmdlet already has `[CmdletBinding()]`, how do you feel like converting that comment into a `Write-Information` call inside the condition? Then if condition is hit while the cmdlet is called with the `-InformationAction Continue` parameter then at least a message appears instead of silently skipping.

Reply:
That sounds good, although I couldn't get it to display when running the analysis normally, but it works if I run this specific cmdlet. Am I missing something?

Reply:
Good question, I'm not sure. That switch should inherit down across calls. What happens if you set the [`$InformationPreference = 'Continue'`](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_preference_variables?view=powershell-7.4#informationpreference)  in the shell before calling it?

Reply:
It doesn't make a difference - `Invoke-ScriptAnalyzer` or `Invoke-Analyzer` is probably swallowing the output.